### ISSUE TITLE
a problem with reproducibility and finetuning

### ISSUE THREAD

**Issue by davidbuhtig** [rbgirshick/py-faster-rcnn#650]
I have been training with faster-rcnn lately and got a problem:
 
I did multiple, different trainings, only to get different contents of ImageSets/Main/trainval.txt and ImageSets/Main/test.txt, i.e. different sequences of the training samples fed into the network. All the other parameters remain the same: training dataset(JPEGImages, Annotations 6k), training iterations(20k), training network(ZF, alt_opt) and so on. Later when testing with these trained models I got significant deviations(with same threshold of 0.7), like difference in number of detected pictures could be close to 100 in a test set of 1000 pictures, while AP values were pretty much the same (around 0.60).  
 
I would like to ask what has caused the deviations? Is it because some training samples are poor or training has not converged?
 
If poor samples are the cause, can I assume that samples that make the smoothed loss curve abrupt are of bad quality?
But if the latter is the case, I wonder why I got worse results sometimes when I increased iterations to 50k or 80k than 20k iterations. 
Or maybe something else lead to this?

This problem makes it difficult for me to reproduce a fine result with different sequences in trainval.txt, or to finetune a net with a trained model and some additional samples.

I have also attached the training loss curve for reference. 
![loss-zf-alt_opt-20000-20170802195556](https://user-images.githubusercontent.com/10024489/29115678-04f0fd44-7d2b-11e7-8a04-3373cbe366dd.jpg)
